magento/pwa-studio
# [PWA-324] [bug]: venia-static files don't load out-of-the box with scaffolding command

## Description

See #2109 

This fix resolves two issues, but has uncovered one more, which will be investigated and resolved in a fast follow.

**Fixes:**
* 500 error of venia static assets when doing `yarn watch` or `yarn start`
* `venia-static` missing from published package

**Known Issues:**
* While using `yarn watch`, webpack is not actually serving the actual assets. Nothing is throwing an error, but something is wrong with the content type, or the headers, or something odd that webpack is doing with its virtual file system. You can navigate directly to assets and they load fine (eg. `/venia-static/favicon.ico`), so this should just be something simple with webpack or UPWARD. To be fixed by https://jira.corp.magento.com/browse/PWA-357.

## Related Issue





* [[PWA-324](https://jira.corp.magento.com/browse/PWA-324)] [bug]: venia-static files don't load out-of-the box with scaffolding command
* Closes #2109

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


_Note:_ The `DEBUG_PROJECT_CREATION` flag below will simulate fetches of package tarballs from npm by running `npm pack` locally and serving those.
1. From a sibling directory to pwa-studio, run the create-pwa binary (`mkdir create-pwa-test; cd create-pwa-test; DEBUG_PROJECT_CREATION=1 ../pwa-studio/packages/create-pwa/bin/create-pwa`
2. Step through setup and install package dependencies
3. `yarn watch`, and verify that 500 errors are no longer being thrown
4. Optionally, you can also go into `node_modules/@magento/venia-ui/` and verify that `venia-static` is now there in the package
## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
